### PR TITLE
Accessibility: Accessible names

### PR DESCRIPTION
**Pull request by seanpdoyle**
Whenever an element can receive focus or is otherwise interactive,
application authors are responsible for providing it with an [accessible
name][].

The [WAI ARIA Authoring Practices Guides][wai-aria] dedicates several
sections to outlining _how_ browsers and assistive technologies derive
an element's name. Instead of attempting to outline those methods with
the same level of detail as that document, this commit highlights (and
links to) the [5 Cardinal Rules of Naming][].

[accessible name]: https://www.w3.org/TR/wai-aria-practices/#names_and_descriptions
[wai-aria]: https://www.w3.org/TR/wai-aria-practices/
[5 Cardinal Rules of Naming]: https://www.w3.org/TR/wai-aria-practices/#naming_cardinal_rules

### CODE REVIEW

**Review by stevehanson — APPROVED** [review]
Thank you! That W3 doc was an informative read.

File: accessibility/README.md [hunk 1]
```diff
@@ -52,6 +52,13 @@ Automated checks can catch a lot of common issues before they reach production.
 
 ### Development
 
+- Ensure every focusable or interactive element has an [accessible name][]
+- Follow the [Cardinal Rules of Naming][]:
+  1. [Heed Warnings and Test Thoroughly][rule-1]
+  2. [Prefer Visible Text][rule-2]
```
Comment (seanpdoyle) [inline accessibility/README.md:58]: Would "Prefer Visible Text" also be a guideline worth including under this document's `Design` heading?

Comment (stevehanson) [reply accessibility/README.md:58]: I think that would make sense. What would you think of something a little more self-explanatory, like:

> Ensure interactive elements [have a text label][rule-2]

Comment (seanpdoyle) [reply accessibility/README.md:59]: Added!

**Review by cpjmcquillan — APPROVED** [review]

**Review by mike-burns — APPROVED** [review]

**Pull request merged**